src/csv.rs

```diff
@@ -0,0 +1,100 @@
+use csv::Reader;
+use serde::Deserialize;
+use std::error::Error;
+use std::fs::File;
+use std::path::Path;
+use serde_with::serde_as;
+
+#[serde_as]
+#[derive(Debug, Deserialize)]
+/// Represents an agent with various attributes including name, description, and decision rules.
+pub struct Agent {
+    /// The name of the agent (required).
+    name: String,
+    /// A description of the agent (optional).
+    description: Option<String>,
+    /// The name of the commodity the agent deals with (required).
+    commodity_name: String,
+    /// The portion of the commodity the agent handles, ranging from 0 to 1 (required).
+    commodity_portion: f32, // f32 is sufficient for values between 0 and 1
```

Comment:
While `f64`s do have a larger range of possible values than `f32`s, they also have higher precision, even between 0 and 1. We probably don't need the precision to be that high, but let's just consistently use `f64`s everywhere for now to make the calculations slightly simpler:

```suggestion
    commodity_portion: f64,
```